text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,153 @@
+
+#### Stable
+
+The component is ready for general availability. Bugs and performance problems should be reported, and there's an expectation that the component owners will work on them. Breaking changes, including configuration options and the component's output, are not expected to happen without prior notice unless under special circumstances.
+
+#### Deprecated
+
+Development of this component is halted, and no further support will be provided. No new versions are planned, and the component might be removed from its included distributions. Note that new issues will likely not be worked on except for critical security issues. Components that are included in distributions are expected to exist for at least two minor releases or six months, whichever happens later. They also MUST communicate in which version they will be removed.
+
+#### Unmaintained
+
+A component identified as unmaintained does not have an active code owner. Such components may have never been assigned a code owner, or a previously active code owner has not responded to requests for feedback within 6 weeks of being contacted. Issues and pull requests for unmaintained components SHOULD be labeled as such. After 6 months of being unmaintained, these components MAY be deprecated. Unmaintained components are actively seeking contributors to become code owners.
+
+### Specification requirements
+
+The specification should declare all requirements for a specified component to be declared stable. When implementations are still missing features required by the specification, they can be at most Alpha status.
+
+### Tier-1 SIGs
+
+This OTEP recommends the following SIGs to be included as part of the Tier-1 SIGs for OpenTelemetry:
+
+* Spec General
+* Semantic Conventions (WG)
+* Collector
+* Java
+* Go
+* .NET
+* JavaScript
+
+Once this OTEP is approved, the TC and GC members will vote on the individual SIGs from the list above and are free to make further suggestions. A final list of Tier-1 SIGs shall be in the "community" repository.
+
+### Tier-1 Components
+
+When discussing the components' maturity, we often debate what should be included in the evaluation criteria. To help clarify that, we introduce the concept of "Tier-1 SIGs" and "Tier-1 Components". Tier-1 Components are critical for a specific SIG: only when all Tier-1 Components are stable can a project be evaluated as stable. Example:
+
+* The "OTLP receiver" for the OpenTelemetry Collector could potentially be a "Tier-1 Component". Only after this component is declared stable may the Collector be declared stable, provided all other Tier-1 Components are also stable.
```

Comment:
I do not understand the motivation here. Why do we need these derivation rules? If, as an end user, I am deploying a collector which includes `otlpreceiver` as `stable` and `config` module as `beta`, what value do I get from any maturity designation at the whole collector level? I think I already have all the info I need at the component level.

Reply:
You are right: as an end-user, I would care only about the collector's calculated stability. I can then assume that all components within the collector are of the same stability or higher.

However, as a maintainer of the project, I care about the stability of the individual components, so that I can make a decision on which components to include in the deliverable.

As a code owner for a component, I care about the stability of my component so that it can be included in a tier-1 deliverable.

Reply:
> You are right: as an end-user, I would care only about the collector's calculated stability.

My point was exactly the opposite: as an end user I *do not* care about cumulative/calculated stability level, because it's largely useless being the lowest common denominator.

Reply:
As a Collector user I don't care about the least stable Collector component if I am not going to use that component. It has no impact on me because by design components that I don't use are not even instantiated at runtime. On the other hand I do care about Collector's core modules of the codebase (e.g. pdata, etc) because they impact every single component, they are foundational dependency of every other component.

So, really it is not simply a minimum stability of all components included, there may be more user-specific nuance and situational nuance depending on what the dependency graph looks like. The calculated "overall" stability may or may not be the minimum of all components. We should probably leave out the calculation rule from the OTEP and let maintainers specify how to assess the "overall stability" if we think it is important to be documented.

Reply:
> When I see that the Core distribution is stable, I assume I can use everything in it and expect it to work. Having a "the Collector is stable, except for the following components..." is a confusing message.

That's why we may want to avoid making any claims about stability of the Collector as a whole, since it is inherently meaningless. Linux has a whole bunch of subsystems of different maturities, does it have a notion of a single, distro-level "Stable"?

Reply:
Not sure about the Kernel (although I remember a time when a pair version was a stable one, and an odd version was experimental), but I consume the kernel as part of a distribution. When I install Fedora or Red Hat Enterprise Linux, I expect it to work out of the box, no matter what the individual versions are. LTS versions of different distributions are quite popular among folks running production-grade workloads because of that: they know that everything in the repositories for that distribution is supported.